test/utils/deployment.ts

```diff
@@ -0,0 +1,121 @@
+    if (yarnStartProcess) {
+        throw new Error('Old 0x-api instance has not been torn down');
+    }
+    await setupDependenciesAsync(loggingConfig.shouldPrintDependencyLogs || false);
+    yarnStartProcess = spawn('yarn', ['start'], {
+        cwd: apiRootDir,
+    });
+    if (loggingConfig.shouldPrintApiLogs) {
+        yarnStartProcess.stdout.on('data', chunk => {
+            neatlyPrintChunk('[0x-api]', chunk);
+        });
+        yarnStartProcess.stderr.on('data', chunk => {
+            neatlyPrintChunk('[0x-api | error]', chunk);
+        });
+    }
+    // Wait for the API to boot up
+    // HACK(jalextowle): This should really be replaced by log-scraping, but it
+    // does appear to work for now.
+    await sleepAsync(5000); // tslint:disable-line:custom-no-magic-numbers
+}
+
+/**
+ * Tears down the old 0x-api instance.
+ * @param loggingConfig Whether or not the logs from the teardown functions should
+ *        be printed.
+ */
+export async function teardownApiAsync(loggingConfig: LoggingConfig = {}): Promise<void> {
+    if (!yarnStartProcess) {
+        throw new Error('There is no 0x-api instance to tear down');
+    }
+    yarnStartProcess.kill();
+    await teardownDependenciesAsync(loggingConfig.shouldPrintDependencyLogs || false);
+}
+
+/**
+ * Sets up 0x-api's dependencies.
+ * @param shouldPrintLogs Whether or not the logs from `docker-compose up`
+ *        should be printed.
+ */
+export async function setupDependenciesAsync(shouldPrintLogs: boolean = false): Promise<void> {
+    const up = spawn('docker-compose', ['up'], {
+        cwd: apiRootDir,
+        env: {
+            ...process.env,
+            ETHEREUM_RPC_URL: 'http://ganache:8545',
+            ETHEREUM_CHAIN_ID: '1337',
+        },
+    });
+    if (shouldPrintLogs) {
+        up.stdout.on('data', chunk => {
+            neatlyPrintChunk('[docker-compose up]', chunk);
+        });
+        up.stderr.on('data', chunk => {
+            neatlyPrintChunk('[docker-compose up | error]', chunk);
+        });
+    }
+    // Wait for the dependencies to boot up.
+    // HACK(jalextowle): This should really be replaced by log-scraping, but it
+    // does appear to work for now.
+    await sleepAsync(10000); // tslint:disable-line:custom-no-magic-numbers
+}
+
+/**
+ * Tears down 0x-api's dependencies.
+ * @param shouldPrintLogs Whether or not the logs from `docker-compose down`
+ *        should be printed.
+ */
+export async function teardownDependenciesAsync(shouldPrintLogs: boolean = false): Promise<void> {
+    const down = spawn('docker-compose', ['down'], {
+        cwd: apiRootDir,
+    });
+    if (shouldPrintLogs) {
+        down.stdout.on('data', chunk => {
+            neatlyPrintChunk('[docker-compose down]', chunk);
+        });
+        down.stderr.on('data', chunk => {
+            neatlyPrintChunk('[docker-compose down | error]', chunk);
+        });
+    }
```

Comment:
So, the `Mesh` instance launched by `docker-compose up` is writing to `0x_mesh` directory. Can we delete at after a run? Additionally, the Postgres DB changes are also persisted to `postgres` I believe.

Reply:
Yep, this was an oversight on my part. 

I may add archival logic so that you don't actually use the state, but you get a clean slate each time.